src/Api/SecretsManager/Controllers/AccessPoliciesController.cs

```diff
@@ -336,4 +341,39 @@ await _accessPolicyRepository.GetServiceAccountGrantedPoliciesPermissionDetailsA
                 userId, accessClient);
         return new ServiceAccountGrantedPoliciesPermissionDetailsResponseModel(results);
     }
+
+    public async Task LogAccessPolicyServiceAccountChanges(IEnumerable<BaseAccessPolicy> currentPolicies, IEnumerable<BaseAccessPolicy> updatedPolicies, Guid userId)
+    {
+        foreach (var current in currentPolicies.OfType<GroupServiceAccountAccessPolicy>())
+        {
+            if (updatedPolicies.Where(r => r.Id == current.Id).Count() <= 0)
+            {
+                await _eventService.LogServiceAccountGroupEventAsync(userId, current, EventType.ServiceAccount_GroupRemoved, _currentContext.IdentityClientType);
```

Comment:
Since we're looping and in the method we're wrapping the bulk method, could instead a command be created and just use the bulk method with all the events needed?